### ISSUE TITLE
jackson should be independent of bom version

### ISSUE THREAD

**Issue by sddakoty** [spring-projects/spring-boot#13173]
Using `spring-boot-starter-parent.1.5.13.RELEASE`:

The following dependency cannot be resolved as it does not exist:
```
	<dependencies>
            <dependency>
                <groupId>com.fasterxml.jackson.datatype</groupId>
                <artifactId>jackson-datatype-jsr310</artifactId>
                <version>${jackson.version}</version> <!-- 2.8.11.20180405 does not exist -->
            </dependency>
	</dependencies>
```

Root cause is that `spring-boot-dependencies` uses `jackson.version` for the `bom` import.

Yet I don't know why spring uses a "custom" version `.20180405` here which cannot be found in the public repository.
Worse: any additional modules like the `jackson-datatype-jsr310` that depend on the parent `jackson.version` property will fail hereafter.

I know the issue can be resolved by just introducing an additional version property for the jackson module. But I'd prefer the module being upgraded when spring-boot-parent upgrades the jackson version. So I'd like to still make use of `jackson.version` here, which is impossible with the custom bom version.

**Comment by snicoll** [comment]
Duplicate of #12790

**Comment by sddakoty** [comment]
So you're saying the solution is: use a different property than `jackson.version` for including the jackson modules? I was hoping I could still reply on the spring-parent dependency versions, like it is for all other dependencies...

**Comment by snicoll** [comment]
@sddakoty no, I don't understand what the problem actually is. Perhaps [this comment will help](https://github.com/spring-projects/spring-boot/issues/12790#issuecomment-383267578)? Not specifying a version at all for these is the right way to proceed.

**Comment by sddakoty** [comment]
Let me rephrase: the problem is that jackson modules like `jackson-datatype-jsr310` cannot be inherited from `spring-boot-starter-parent` without using a version number explicit. Eg the following does not resolve the module:
```
<project>
    <parent>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-starter-parent</artifactId>
        <version>1.5.13.RELEASE</version>
    </parent>

    <dependencies>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-web</artifactId>
            <exclusions>
                <exclusion>
                    <groupId>org.springframework.boot</groupId>
                    <artifactId>spring-boot-starter-logging</artifactId>
                </exclusion>
            </exclusions>
        </dependency>

	<dependency>
            <groupId>com.fasterxml.jackson.datatype</groupId>
            <artifactId>jackson-datatype-jsr310</artifactId>
        </dependency>
    </dependencies>
</project>
```

Thus, one has to add the version explicit. Now it would be handy to just use `${jackson.version}`, inheriting the spring-boot defined jackson version from parent. As on spring-upgrade, also the modules would automatically be upgraded.

BUT `jackson.version` is `2.8.11.20180405`, which does not exist for the `jackson-datatype-jsr310` artifact (only 2.8.11 would exist).

Therefore with the current design it's impossible to inherit the jackson defined version from spring parent. Of course I could introduce my own `<jackson-module.version>2.8.11</jackson-module.version>` (that is also suggested in your linked comment). But it would be nice if one could just reuse/inherit the spring parent version...

**Comment by wilkinsona** [comment]
> the problem is that jackson modules like jackson-datatype-jsr310 cannot be inherited from spring-boot-starter-parent

I don't believe that's the case.

When one bom is imported into another, Maven effectively inlines the imported bom into the importing bom. The Jackson 2.8.11.20180405 bom manages the `jackson-datatype-jsr310` module with a version of `2.8.11`. This means that the importing bom ends up with a dependency management entry like this:

```xml
<dependency>
    <groupId>com.fasterxml.jackson.datatype</groupId>
    <artifactId>jackson-datatype-jsr310</artifactId>
    <version>2.8.11</version>
</dependency>
```

If you inherit from the importing bom (as you are doing by using `spring-boot-starter-parent` as your project's parent) you can then depend on `jackson-datatype-jsr310` without a version.

If you have a project that does not behave in the manner described above, please share it with us as a repository on GitHub or a zip attached to this issue and we'll take a look.
